pkg/entrypoint/server_test.go

```diff
@@ -0,0 +1,197 @@
+package entrypoint
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap"
+)
+
+const apiAddr = "localhost:8080"
+
+type testCase struct {
+	input    []Job
+	expected []statusResponse
+}
+type statusResponse struct {
+	Jobs []job
+}
+type job struct {
+	Name   string
+	Status string
+}
+
+var log logr.Logger
+var _ = BeforeSuite(func() {
+	zapLog, _ := zap.NewDevelopment()
+	log = zapr.NewLogger(zapLog)
+})
+
+var _ = Describe("entrypoint status API test", func() {
+	It("should state of successful command changed from running to completed", func() {
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "sleep",
+					Args:    []string{"1"},
+				},
+			},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Running"}}},
+				{Jobs: []job{{Name: "test1", Status: "Completed"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+
+	It("should state of command with exit code 1 become failed", func() {
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "false",
+					Args:    []string{},
+				}},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Failed"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+
+	It("should state of unknown command become failed", func() {
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "unknowncommand",
+					Args:    []string{},
+				}},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Failed"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+
+	It("should state of not permission command become failed", func() {
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "./testresources/testscript.sh",
```

Comment:
It would be nice to rename this script to something that explicitly indicate that it does not have execute permission.